### DISCUSSION TITLE
I think spaceauth is not useful enough for CI servers

### DISCUSSION THREAD

**Discussion by turkcankeskin** [fastlane/fastlane#22186] [Ideas]
Hello,

What I want to say is; fastlane spaceauth does not work useful enough for CI servers.

1- Why?
In my company, when I send my ipa file with fastlane upload_to_testflight, my developer teams want to send the "changelog" value. However, I cannot directly do this due to Apple's 2FA system. Because my Jenkins pipeline expects the 2FA 6-digit code in the fastlane upload step and gets an error.

Fastlane spaceauth's documentation states that the cookie will be stored for 30 days, but this is old information. I observed that the renewal of Apple's policies reduced this period to 30 minutes. (In my case, my cookie is usually exposed after 30 minutes.)

The suggestions are not very useful; because all enterprises know the harms of removing 2FA and security teams do not allow it.

2- Fastlane spaceauth's terminal usage is not suitable for CI pipeline. I think we need a change here.
```
fastlane spaceauth -u email@email.com -p **password** -y
-u = this is already available
-p = here FASTLANE_PASSWORD meets the need
-y = we need to transmit fastlane session directly in a single command line.
```

![image](https://github.com/user-attachments/assets/a05c6a94-a574-4e95-83e4-1053cf10e5ec)

3- My suggestion for 2FA
If the user you released to testflight via fastlane is defined on your CI server, this 2FA code will appear directly on your CI server screen as a popup. With this code, I can get the 2FA code on the popup directly to the terminal.

![image](https://github.com/user-attachments/assets/30c8781e-913c-48f4-a80e-96f58d9b49a0)

```
  #! /usr/bin/env osascript
  tell application "System Events"
      if name of every process contains "FollowUpUI" then
          tell window 1 of process "FollowUpUI"
              click button "Allow"
              delay 2
              set code to value of static text 1 of group 1
              log (code)
              click button "Done"
          end tell
      else
          log ("Couldn't find 2FA window")
      end if
  end tell
  return
```

Here, fastlane's spaceauth can run this for us in the background and provide this directly in our CI pipeline.

`
fastlane spaceauth -u email@email.com -p **password** -y -2fa(?)
`
I did this in python to connect to developer apple. But I'm not doing it in fastlane right now because the question "sould fastlane copy the cookie into your clipboard, so you can easily paste it?" stops my CI pipeline. and I don't get a popup without giving my email address and password with fastlane -u.

**Comment by gregersn** [comment]
At the very least the question about copying to clipboard should be removed. It is also annoying with manual use. Espescially when it is something that is run many times a day on multiple servers.
